subprojects/core-api/src/main/java/org/gradle/api/file/CopyProcessingSpec.java

```diff
@@ -122,6 +124,42 @@ public interface CopyProcessingSpec extends ContentFilterable {
      */
     CopyProcessingSpec setDirMode(@Nullable Integer mode);
 
+    /**
+     * Property for configuring file access permissions.
+     * For details see {@link FilePermissions}.
+     *
+     * @since 8.3
+     */
+    @Incubating
+    Property<FilePermissions> getFilePermissions();
+
+    /**
+     * Configuration action for specifying file access permissions.
+     * For details see {@link FilePermissions}.
+     *
+     * @since 8.3
+     */
+    @Incubating
+    CopyProcessingSpec filePermissions(Action<? super FilePermissions> configureAction);
```

Comment:
💭  We really shouldn't add these methods manually. They should be generated instead. Not something to fix in this PR, but we should address this in a more general way.